com.unity.sg2/Editor/GraphUI/DataModel/Constants/BaseShaderGraphConstant.cs

```diff
@@ -7,31 +7,54 @@
 
 namespace UnityEditor.ShaderGraph.GraphUI
 {
+    //public bool TryGetNodeReader(out NodeHandler reader)
+    //{
+    //    try
+    //    {
+    //        if (graphDataName == null)
+    //        {
+    //            reader = registry.GetDefaultTopology(m_PreviewRegistryKey);
+    //            return true;
+    //        }
+
+    //        reader = graphHandler.GetNode(graphDataName);
+
+    //        return reader != null;
+    //    }
+    //    catch (Exception exception)
+    //    {
+    //        AssertHelpers.Fail("Failed to retrieve node due to exception:" + exception);
+    //        reader = null;
+    //        return false;
+    //    }
+
+
     [Serializable]
-    public abstract class BaseShaderGraphConstant : IConstant, ISerializationCallbackReceiver
+    public abstract class BaseShaderGraphConstant : IConstant
     {
+        [SerializeReference]
+        protected ShaderGraphModel graphModel;
         [SerializeField]
-        private object tempSerializedValue;
-
-        protected GraphHandler graphHandler;
-
+        protected string nodeName;
         [SerializeField]
-        protected string nodeName, portName;
+        protected string portName;
+        GraphHandler graphHandler => graphModel.GraphHandler;
+
         public bool IsInitialized => !string.IsNullOrEmpty(nodeName) && graphHandler != null;
         public FieldHandler GetField()
         {
             if (!IsInitialized) return null;
-            var nodeReader = graphHandler.GetNode(nodeName);
+            var nodeReader = graphHandler.GetNode(nodeName) ?? graphModel.RegistryInstance.defaultTopologies.GetNode(nodeName);
             var portReader = nodeReader.GetPort(portName);
             return portReader.GetTypeField();
         }
```

Comment:
Needed to accommodate node as searcher preview cases-- there is definitely a better way to do this (but would want to involve changing how we populate the searcher in general).

Reply:
It would be nice if we just had a separate model used for the searcher maybe-- that way the default topologies can be passed into it as the graph handler, so all of the nodes are working from that, instead of having a special case in a few places. But IDK- it's fine for now.